### PR TITLE
Fix git diff bug

### PR DESCRIPTION
**Pull request by Carlgo11**
This PR removes the white space on git diff lines. This in turn fixes recent build failures that where due to `cut` being incorrectly used to remove white space.

I also reformatted  twitter.rb.

### CODE REVIEW

**Comment by phallobst** [comment]
Something like
```rb
domains = `diff ... | sed -n 's/.*"domain"[^"]*"\\(.*\\)".*/\\1/p'`
domains.split("\n").each do |domain|
  ...
end
```
is shorter and faster, and you need the key pattern only once. (same for `facebook` and `twitter`)

File: tests/alexa.rb [hunk 1]
```diff
@@ -41,14 +41,14 @@ def store_cache(site, rank)
 
 status = 0
 # Fetch changes
-diff = `git diff origin/master...HEAD entries/ | grep "^+[[:space:]]*\\"domain\\":" | cut -c17-`
+# rubocop:disable Layout/LineLength
+diff = `git diff origin/master...HEAD entries/ | grep "^+[[:space:]]*\\"domain\\":" | sed -n 's/.*"domain"[^"]*"\\(.*\\)".*/\\1/p'`
+# rubocop:enable Layout/LineLength
 # Strip and loop through diff
 diff.gsub("\n", '').split('",').each do |site|
```
Comment (phallobst) [inline tests/alexa.rb]:
```suggestion
diff.split('\n').each do |site|
```

File: tests/alexa.rb [hunk 2]
```diff
@@ -41,14 +41,14 @@ def store_cache(site, rank)
 
 status = 0
 # Fetch changes
-diff = `git diff origin/master...HEAD entries/ | grep "^+[[:space:]]*\\"domain\\":" | cut -c17-`
+# rubocop:disable Layout/LineLength
+diff = `git diff origin/master...HEAD entries/ | grep "^+[[:space:]]*\\"domain\\":" | sed -n 's/.*"domain"[^"]*"\\(.*\\)".*/\\1/p'`
```
Comment (phallobst) [inline tests/alexa.rb]:
```suggestion
diff = `git diff origin/master...HEAD entries/ | sed -n 's/^+.*"domain"[^"]*"\\(.*\\)".*/\\1/p'`
```

File: tests/facebook.rb [hunk 3]
```diff
@@ -5,7 +5,9 @@
 require 'uri'
 
 status = 0
-diff = `git diff origin/master...HEAD entries/ | grep "^+[[:space:]]*\\"facebook\\":" | cut -c21-`
+# rubocop:disable Layout/LineLength
+diff = `git diff origin/master...HEAD entries/ | grep "^+[[:space:]]*\\"facebook\\":" | sed -n 's/.*"facebook"[^"]*"\\(.*\\)".*/\\1/p'`
```
Comment (phallobst) [inline tests/facebook.rb]:
```suggestion
diff = `git diff origin/master...HEAD entries/ | sed -n 's/^+.*"facebook"[^"]*"\\(.*\\)".*/\\1/p'`
```

File: tests/facebook.rb [hunk 4]
```diff
@@ -5,7 +5,9 @@
...
+# rubocop:enable Layout/LineLength
 diff.gsub("\n", '').gsub(',', '').split('"').each do |page|
```
Comment (phallobst) [inline tests/facebook.rb]:
```suggestion
diff.split('\n').each do |page|
```

File: tests/twitter.rb [hunk 5]
```diff
@@ -11,20 +11,20 @@
 end
 
 status = 0
-diff = `git diff origin/master...HEAD entries/ | grep "^+[[:space:]]*\\"twitter\\":" | cut -c20-`
+# rubocop:disable Layout/LineLength
+diff = `git diff origin/master...HEAD entries/ | grep "^+[[:space:]]*\\"twitter\\":" | sed -n 's/.*"twitter"[^"]*"\\(.*\\)".*/\\1/p'`
```
Comment (phallobst) [inline tests/twitter.rb]:
```suggestion
diff = `git diff origin/master...HEAD entries/ | sed -n 's/^+.*"twitter"[^"]*"\\(.*\\)".*/\\1/p'`
```

File: tests/twitter.rb [hunk 6]
```diff
@@ -11,20 +11,20 @@
...
+# rubocop:enable Layout/LineLength
 diff.gsub("\n", '').split('"').each do |handle|
```
Comment (phallobst) [inline tests/twitter.rb]:
```suggestion
diff.split('\n').each do |handle|
```

**Review by phallobst — CHANGES_REQUESTED** [review]
I guess I wasn't clear enough with my suggestion.

**Review by phallobst — APPROVED** [review]

File: tests/facebook.rb [hunk 7]
```diff
@@ -5,8 +5,8 @@
 require 'uri'
 
 status = 0
-diff = `git diff origin/master...HEAD entries/ | grep "^+[[:space:]]*\\"facebook\\":" | cut -c21-`
-diff.gsub("\n", '').gsub(',', '').split('"').each do |page|
+diff = `git diff origin/master...HEAD entries/ | sed -n 's/^+.*"facebook"[^"]*"\\(.*\\)".*/\\1/p'`
+diff.split('\n').each do |page|
```
Comment (ApCoder123) [inline tests/facebook.rb]:
```suggestion
diff.split(' ').each do |page|
```
I might be missing something here, but when I run this, I get the error:
```
bad URI(is not URI?): "https://www.facebook.com/pg/myinvalidfacebookpage\\n" (URI::InvalidURIError)
```
There seems to be a `\n` that isn't getting removed. The test works when splitting at new lines is changed to splitting at spaces or removing the `\n` from the string.

Comment (Carlgo11) [reply tests/facebook.rb]: Nice catch.

I have no clue why that would be the case. The variable `diff` is literally `rbc\nvygruppen\n`. No spaces, just linebreakers. Splitting with `' '` does indeed fix it though.
Maybe @phallobst knows what is happening?

**Comment by phallobst** [comment]
It seems that `foo.split('\n')` is not the same as `foo.split("\n")` - the double quotes are needed for the line break processing (or `foo.split(/\n/)` as an alternative).

Also, I've learned that a single space for the pattern means "any whitespace character", so that works with line breaks as well.

Who designed this? :vomiting_face:

File: tests/alexa.rb [hunk 8]
```diff
@@ -41,9 +41,9 @@ def store_cache(site, rank)
 
 status = 0
 # Fetch changes
-diff = `git diff origin/master...HEAD entries/ | grep "^+[[:space:]]*\\"domain\\":" | cut -c17-`
+diff = `git diff origin/master...HEAD entries/ | sed -n 's/^+.*"domain"[^"]*"\\(.*\\)".*/\\1/p'`
 # Strip and loop through diff
-diff.gsub("\n", '').split('",').each do |site|
+diff.split('\n').each do |site|
```
Comment (phallobst) [inline tests/alexa.rb]:
```suggestion
diff.split("\n").each do |site|
```

File: tests/facebook.rb [hunk 9]
```diff
@@ -5,8 +5,8 @@
 require 'uri'
 
 status = 0
-diff = `git diff origin/master...HEAD entries/ | grep "^+[[:space:]]*\\"facebook\\":" | cut -c21-`
-diff.gsub("\n", '').gsub(',', '').split('"').each do |page|
+diff = `git diff origin/master...HEAD entries/ | sed -n 's/^+.*"facebook"[^"]*"\\(.*\\)".*/\\1/p'`
+diff.split(' ').each do |page|
```
Comment (phallobst) [inline tests/facebook.rb]:
```suggestion
diff.split("\n").each do |page|
```

File: tests/twitter.rb [hunk 10]
```diff
@@ -11,12 +11,14 @@
 end
 
 status = 0
-diff = `git diff origin/master...HEAD entries/ | grep "^+[[:space:]]*\\"twitter\\":" | cut -c20-`
-diff.gsub("\n", '').split('"').each do |handle|
+diff = `git diff origin/master...HEAD entries/ | sed -n 's/^+.*"twitter"[^"]*"\\(.*\\)".*/\\1/p'`
+diff.split('\n').each do |handle|
```
Comment (phallobst) [inline tests/twitter.rb]:
```suggestion
diff.split("\n").each do |handle|
```

**Review by phallobst — CHANGES_REQUESTED** [review]

**Review by phallobst — APPROVED** [review]

**Review by ApCoder123 — APPROVED** [review]

**Pull request merged**